test/extended/openstack/cloud-provider-config.go

```diff
@@ -104,6 +104,70 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The Openshift", f
 				}
 			}
 		})
+
+		//https://bugzilla.redhat.com/show_bug.cgi?id=2074471
+		g.It("should update use-octavia property depending on the NetworkType", func() {
+
+			userConfigMap := config_params{
+				namespace: "openshift-config",
+				name:      "cloud-provider-config",
+				key:       "config",
+				skip:      nil,
+			}
+
+			ccmConfigMap := config_params{
+				namespace: "openshift-cloud-controller-manager",
+				name:      "cloud-conf",
+				key:       "cloud.conf",
+				skip:      nil,
+			}
+
+			section_name := "LoadBalancer"
+			property_name := "use-octavia"
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			c, err := configv1client.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			user_cfg, err := getConfig(clientSet,
+				userConfigMap.namespace,
+				userConfigMap.name,
+				userConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ccm_cfg, err := getConfig(clientSet,
+				ccmConfigMap.namespace,
+				ccmConfigMap.name,
+				ccmConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			user_property_value, err := getPropertyValue(section_name, property_name, user_cfg)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+				"%q (namespace: %q)", property_name, section_name, userConfigMap.name, userConfigMap.namespace)
+
+			ccm_property_value, err := getPropertyValue(section_name, property_name, ccm_cfg)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+				"%q (namespace: %q)", property_name, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+
+			e2e.Logf("Checking config for networkType detected: %q", network.Status.NetworkType)
+			switch network.Status.NetworkType {
+			case "Kuryr":
+				o.Expect(ccm_property_value).Should(o.Equal("true"),
```

Comment:
Wait, I have an impression it should be "false" when Kuryr is running. Yet this passes tests, so I'm confused now.

Reply:
Yeah, seems like test does nothing.

Reply:
According to the logs I am writting and the different tests I did, it is working fine:

```
$ make openstack-tests && KUBECONFIG=~/ostest/auth/kubeconfig ./openstack-tests run-test "[sig-installer][Suite:openshift/openstack] The Openshift on cloud provider configuration should update use-octavia and enabled properties depending on the NetworkType [Suite:openshift/conformance/parallel]"
go build -o openstack-tests ./cmd/openshift-tests
[BeforeEach] [Top Level]
  /home/stack/openstack-test/vendor/github.com/openshift/origin/test/extended/util/test.go:61
[BeforeEach] on cloud provider configuration
  /home/stack/openstack-test/test/extended/openstack/cloud-provider-config.go:31
STEP: preparing openshift dynamic client
[It] should update use-octavia and enabled properties depending on the NetworkType [Suite:openshift/conformance/parallel]
  /home/stack/openstack-test/test/extended/openstack/cloud-provider-config.go:109
Jul 26 16:15:54.032: INFO: Checking config for networkType detected: "Kuryr"
Jul 26 16:15:54.032: INFO: Properties with correct values on section "LoadBalancer" in configMap "cloud-conf" (namespace: "openshift-cloud-controller-manager") with NetworkType=Kuryr.
Jul 26 16:15:54.032: INFO: - Properties (["use-octavia" "enabled"]): Values (["true" "false"])

```

Reply:
the ccm config map is setting use-octavia=True and enabled=False. This is what I understood from the BZ and the PR fixing it (https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/183)